src/test/java/com/google/step/servlets/VoteServletTest.java

```diff
@@ -163,52 +162,4 @@ public void testDoGet_success() throws IOException {
     writer.flush();
     assertEquals("1", stringWriter.toString().trim());
   }
-
-  @Test
-  public void testDoPost_successUserChangedDirection() throws IOException {
-    setUpUserAuthentication();
-    HttpServletRequest request = mock(HttpServletRequest.class);
-
-    when(request.getParameter("dir")).thenReturn(DIR_ONE);
-    when(request.getPathInfo()).thenReturn(DEAL_PATH);
-    when(voteManager.getDirection(USER_ID_A, DEAL_ID_A)).thenReturn(-1);
-
-    servlet.doPost(request, response);
-
-    verify(response).setStatus(HttpServletResponse.SC_ACCEPTED);
-    verify(voteManager).vote(eq(USER_ID_A), eq(DEAL_ID_A), eq(1));
-    verify(dealVoteCountManager).updateDealVotes(eq(DEAL_ID_A), eq(2));
-  }
-
-  @Test
-  public void testDoPost_successUserChangedAnotherDirection() throws IOException {
-    setUpUserAuthentication();
-    HttpServletRequest request = mock(HttpServletRequest.class);
-
-    when(request.getParameter("dir")).thenReturn(DIR_NEG_ONE);
-    when(request.getPathInfo()).thenReturn(DEAL_PATH);
-    when(voteManager.getDirection(USER_ID_A, DEAL_ID_A)).thenReturn(1);
-
-    servlet.doPost(request, response);
-
-    verify(response).setStatus(HttpServletResponse.SC_ACCEPTED);
-    verify(voteManager).vote(eq(USER_ID_A), eq(DEAL_ID_A), eq(-1));
-    verify(dealVoteCountManager).updateDealVotes(eq(DEAL_ID_A), eq(-2));
-  }
-
-  @Test
-  public void testDoPost_successUserUndoesVote() throws IOException {
-    setUpUserAuthentication();
-    HttpServletRequest request = mock(HttpServletRequest.class);
-
-    when(request.getParameter("dir")).thenReturn(DIR_UNDO);
-    when(request.getPathInfo()).thenReturn(DEAL_PATH);
-    when(voteManager.getDirection(USER_ID_A, DEAL_ID_A)).thenReturn(1);
-
-    servlet.doPost(request, response);
-
-    verify(response).setStatus(HttpServletResponse.SC_ACCEPTED);
-    verify(voteManager).vote(eq(USER_ID_A), eq(DEAL_ID_A), eq(0));
-    verify(dealVoteCountManager).updateDealVotes(eq(DEAL_ID_A), eq(-1));
-  }
```

Comment:
Why are these tests deleted? Don't you still need to test the following cases:

1. No vote existed before from the user for this deal.
2. A vote in a different direction existed.
3. A vote in the same direction existed (in which case, VotingHelper shouldn't be called).

Reply:
I transferred the tests to VotingHelperTest instead since i thought the servlet should just be used to test that the votinghelper method was called